src/main/java/storage/Database.kt

```diff
@@ -0,0 +1,155 @@
+import org.intellij.lang.annotations.Language
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.sql.PreparedStatement
+import java.sql.SQLException
+import java.sql.Statement
+import java.sql.Timestamp
+import java.time.Instant
+import kotlin.system.exitProcess
+
+object Database {
+    private val postgres = Postgres()
+    val logger: Logger = LoggerFactory.getLogger(Database::class.java)
+
+    init {
+        doMigration(
+            "CREATE TABLE IF NOT EXISTS questions (id SERIAL PRIMARY KEY, author int8, question varchar(255), answered boolean default false)",
+            "Created questions table."
+        )
+        doMigration(
+            "CREATE TABLE IF NOT EXISTS suggestions (id SERIAL PRIMARY KEY, author int8, question varchar(255))",
+            "Created suggestions table."
+        )
+        doMigration(
+            "ALTER TABLE questions ADD COLUMN IF NOT EXISTS approved_at TIMESTAMP NOT NULL DEFAULT now()",
+            "Added approved_at column to questions table."
+        )
+    }
+
+    private fun doMigration(@Language("SQL") sql: String, infoString: String) {
+        if (!postgres.execute(sql)) {
+            logger.error("Failed to execute everything in setup. Exiting.")
+            exitProcess(1)
+        }
+        logger.info(infoString)
+    }
+
+    fun addSuggestion(author: Snowflake, question: String?): Int {
+        return postgres.insert(
+            "INSERT INTO public.suggestions (author, question) VALUES (?, ?)",
+            SqlConsumer { statement: PreparedStatement ->
+                statement.setLong(1, author.asLong())
+                statement.setString(2, question)
+            }
+        )
+    }
+
+    fun moveSuggestionToQuestion(suggestionId: Int): Int {
+        try {
+            postgres.connection.use { connection ->
+                val author: Long
+                val question: String
+                val querySuggestion = connection.prepareStatement("SELECT * FROM public.suggestions WHERE id = ?")
+                querySuggestion.setInt(1, suggestionId)
+                querySuggestion.executeQuery().use { resultSet ->
+                    if (!resultSet.next()) {
+                        logger.error("Attempted reading non-existant suggestions id {}.", suggestionId)
+                        return -1
+                    }
+                    author = resultSet.getLong("author")
+                    question = resultSet.getString("question")
+                }
+
+                connection.autoCommit = false
+
+                val insertStatement = connection.prepareStatement(
+                    "INSERT INTO public.questions (author, question, approved_at) VALUES (?, ?, ?)",
+                    Statement.RETURN_GENERATED_KEYS,
+                )
+                insertStatement.setLong(1, author)
+                insertStatement.setString(2, question)
+                insertStatement.setTimestamp(3, Timestamp.from(Instant.now()))
```

Comment:
This isn't necessary

Reply:
normally this isnt needed, but i like to have this kind of stuff as a safeguard against weird scenarios where the db server has a different time zone than the jvm. if the app writes the timestamp itself, it will always be correct on read. if the timestamp is auto filled by the db server, you could potentially read a slightly wrong value, based on your configuration ^^

should i leave it in or remove it?